share/ipld/nmt.go

```diff
@@ -95,7 +92,7 @@ func NewNMTNode(id cid.Cid, l, r []byte) ipld.Node {
 }
 
 func (n nmtNode) RawData() []byte {
-	return append([]byte{nmt.NodePrefix}, append(n.l, n.r...)...)
+	return append(n.l, n.r...)
```

Comment:
What about that optimization?😉 

I feel like you had another one on your mind, but don't remember which one exactly.

Reply:
Added another commit with allocation optimisation.